btree/immutable/cacher.go

```diff
@@ -0,0 +1,114 @@
+package btree
+
+import (
+	"sync"
+	"time"
+
+	"github.com/Workiva/go-datastructures/futures"
+)
+
+// cacher provides a convenient interface for retrieving,
```

Comment:
`cacher` is not quite an interface... Maybe you could say in this docstring that it's added threadsafety and caching around a Persister?